platform/orion/core/generic/vault/vault.go

```diff
@@ -241,6 +241,24 @@ func (db *Vault) SetBusy(txid string) error {
 	return nil
 }
 
+func (db *Vault) SetStatus(txID string, code odriver.ValidationCode) error {
```

Comment:
Once we merge the race-conditions PR, we need to check that the error is not caused by a unique-constraint violation.